### PR TITLE
added some functionality that I needed (some inspired from THREE.js)

### PR DESCRIPTION
**Pull request by RobertLRead**
Dear Mr. Stokes: Thanks for an awesome, tiny, functional vector library; I used it to replace THREE.js in a project (https://github.com/PubInv/TriadBalance) for which THREE.js was too heavy.  I needed a few more features, including "clamp" feature inspired from THREE.js. I have added them, attempting to follow your style. There are tests.

### CODE REVIEW

**Comment by francisrstokes** [comment]
Hey Robert,
Thanks for not only the interest but also the contribution! These changes are indeed in the spirit of this library and I'm happy to merge them. I'm not by my primary machine right now, but when I am I'll integrate them and create a release.

File: README.md [hunk 1]
```diff
@@ -74,6 +80,12 @@ and import or require as needed. If you need to use a standalone windowed versio
 
 `vec.dot(v, v2)` - gets dot product of `v` and `v2`
 
+`vec.perpdot(v, v2)` - the perpendicular dot product of `v` and `v2` (sometimes called cross)
```
Comment (francisrstokes) [inline README.md]: I think this should be `perpDot` right?

File: README.md [hunk 2]
```diff
@@ -74,6 +80,12 @@ and import or require as needed. If you need to use a standalone windowed versio
...
+
+`vec.trinagleArea(a, b, c)` - signed area of triangle abc
```
Comment (francisrstokes) [inline README.md]: `triangleArea` 😄

File: README.md (hunk 1 above)
Comment (RobertLRead) [reply README.md]: Fixed.

File: README.md (hunk 2 above)
Comment (RobertLRead) [reply README.md]: Fixed.

**Comment by RobertLRead** [comment]
Dear Mr. Stokes,

I'm angry at myself for not seeing your requests sooner; somehow I did not get notifications and did not check.  I have made the changes that you requested; thank you.  Hopefully this is ready for you now, or for additional review.

**Comment by francisrstokes** [comment]
Done! Thanks @RobertLRead

**Pull request merged**